### PR TITLE
add health check event

### PR DESCRIPTION
**Pull request by silviokennecke**
### 1. Why is this change necessary?
When running Shopware in a docker container, you might want to do some customized health checks.

### 2. What does this change do, exactly?
This PR adds an event to the `HealthCheckController`, which allows developers to add their own health checks without having to decorate the controller or introduce their own endpoint.

### 5. Checklist

- [X] I have rebased my changes to remove merge conflicts
- [X] I have written tests and verified that they fail without my change
- [X] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-implement-new-changelog.md) with all necessary information about my changes
- [X] I have read the contribution requirements and fulfil them.

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at d0b03ca</samp>

This pull request adds a new event system for the health check feature of the API, allowing other components to modify the health check response. It creates a new `HealthCheckEvent` class, injects the `event_dispatcher` service into the `HealthCheckController` class, and adds a changelog entry and a test class for the changes.

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at d0b03ca</samp>

*  Add a new event for health check requests ([link](https://github.com/shopware/shopware/pull/3467/files?diff=unified&w=0#diff-13fad37f684116c48381878e5ab32a0b38b12b97220f474a044a29e6352a1378R1-R9))
   * Create a `HealthCheckEvent` class that extends `Event` and has a `Context` property ([link](https://github.com/shopware/shopware/pull/3467/files?diff=unified&w=0#diff-8f7ffa2cdb70e3a0a484605c817b6bbe90a85fda0ba4135bd6ef876292153c2fR1-R24))
   * Inject the `event_dispatcher` service into the `HealthCheckController` class ([link](https://github.com/shopware/shopware/pull/3467/files?diff=unified&w=0#diff-b9586e2939cf7df11f8e88de86905d8caa9cc0380d48e992427ca8f1712e8401L5-R8), [link](https://github.com/shopware/shopware/pull/3467/files?diff=unified&w=0#diff-756e7ff306bf2376c4506427ce8eae6dc49df25ad76c9c80e6f5887efb4faaf1R59-R62))
   * Modify the `check` method of the `HealthCheckController` to accept a `Context` parameter, create and dispatch a `HealthCheckEvent` with the context, and return a response ([link](https://github.com/shopware/shopware/pull/3467/files?diff=unified&w=0#diff-b9586e2939cf7df11f8e88de86905d8caa9cc0380d48e992427ca8f1712e8401R16-R20), [link](https://github.com/shopware/shopware/pull/3467/files?diff=unified&w=0#diff-b9586e2939cf7df11f8e88de86905d8caa9cc0380d48e992427ca8f1712e8401L18-R30))
*  Add tests for the `check` method of the `HealthCheckController` class ([link](https://github.com/shopware/shopware/pull/3467/files?diff=unified&w=0#diff-1b45d0cea2a1a1505221f36450da58ef3d5a4a4bcf2a42456b2313c7fcf8f74fR1-R50))

### CODE REVIEW

**Comment by AydinHassan** [comment]
@silviokennecke thanks for your contribution, seems like a good extension point. Could you please move the test to `tests/integration/Core/Framework/Api/Controller` ? - the current folder is for legacy tests and new tests should not be added there

File: src/Core/Framework/Api/HealthCheck/Event/HealthCheckEvent.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\Log\Package;
+use Symfony\Contracts\EventDispatcher\Event;
+
+#[Package('core')]
+class HealthCheckEvent extends Event
+{
+    public function __construct(
+        private readonly Context $context
```
Comment (AydinHassan) [inline src/Core/Framework/Api/HealthCheck/Event/HealthCheckEvent.php:16]: We could just make this public then it's not necessary to have a getter

File: src/Core/Framework/Test/Api/Controller/HealthCheckControllerTest.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    {
+        $this->getBrowser()->request('GET', '/api/_info/health-check');
+        $response = $this->getBrowser()->getResponse();
+
+        static::assertEquals(200, $response->getStatusCode());
+    }
+
+    public function testHealthCheckEventIsDispatched(): void
+    {
+        $listener = $this->getMockBuilder(CallableClass::class)->getMock();
```
Comment (AydinHassan) [inline src/Core/Framework/Test/Api/Controller/HealthCheckControllerTest.php:27]: You could use `\Shopware\Core\Test\Stub\EventDispatcher\AssertingEventDispatcher` - see `\Shopware\Tests\Integration\Core\Content\Media\Infrastructure\Path\MediaLocationBuilderTest::testBuildMedia` for an example

**Comment by silviokennecke** [comment]
@AydinHassan I just moved the test and made the context `public readonly`

**Comment by mitelg** [comment]
@silviokennecke it is already merged with https://github.com/shopware/shopware/commit/35e0509e9c601efac600a204eec1da043ad43b49

thanks for your contribution 👍🎉💙

**Pull request closed without merging**